src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs

```diff
@@ -325,6 +325,10 @@ private void LogRequest(string methodName, string?[] providedParameters, Paramet
                     if (providedParameter.StartsWith('[') || providedParameter.StartsWith('{'))
                     {
                         executionParam = _serializer.Deserialize(new JsonTextReader(new StringReader(providedParameter)), paramType);
+                        if (executionParam is null && !IsNullableParameter(expectedParameter))
+                        {
+                            throw new JsonReaderException($"Cannot deserialize parameter `{providedParameter}` into type {paramType}");
+                        }
```

Comment:
Main change: if we try to deserialize an aggregate type like an array or an object and we get `null`, fail if the parameter is not nullable.

Unfortunately, our deserializer does not to provide enough information to differentiate deserializing `null` from failing to deserialize.